src/DIRAC/DataManagementSystem/Agent/RequestOperations/test/Test_ReplicateAndRegister.py

```diff
@@ -0,0 +1,692 @@
+""" That is probably one of the most convoluted unit test file we have...
```

Comment:
If in few years time I will need/want to use a mock of DataManager, I will use/modify add what is in `mock_DM` because ... it is supposed to be a mock of DataManager. And of course by that time I will have completely forgotten the existence of, e.g., function `mock_DM_getReplicas` that you added here. So, I would suggest to move all the mock part over there, and leave in this test module only the tests.

Reply:
but what you want to do in the mock depends on what you test. So I have a hard time imagining what a common DM mock would look like

Reply:
Yes, and all these cases are in the same test file. 
I can see how you factorize utilities like `generateDIRACConfig` (https://github.com/DIRACGrid/DIRAC/blob/ffbe6f7fec889dd0cd9bcbb5e7bdfafa6397705f/src/DIRAC/tests/Utilities/utils.py#L46) but not how you can factorize your mock. Factorizing a mock that works for all tests basically means writing the `monkeypatch` library